lightning/src/ln/channel.rs

```diff
@@ -4618,7 +4624,18 @@ impl<SP: Deref> Channel<SP> where
 			channel_id: Some(self.context.channel_id()),
 		};
 
-		self.context.holder_commitment_point.advance(&self.context.holder_signer, &self.context.secp_ctx, logger);
+		if self.context.holder_commitment_point.advance(&self.context.holder_signer, &self.context.secp_ctx, logger).is_err() {
+			// We only fail to advance our commitment point/number if we're currently
+			// waiting for our signer to unblock and provide a commitment point.
+			// During post-funding channel operation, we only advance our point upon
+			// receiving a commitment_signed, and our counterparty cannot send us
+			// another commitment signed until we've provided a new commitment point
+			// in revoke_and_ack, which requires unblocking our signer and completing
+			// the advance to the next point. So if we get here, either we've messed
+			// up, or our counterparty has.
+			debug_assert!(false, "We should be ready to advance our commitment point by the time we receive commitment_signed");
```

Comment:
hmm, i think maybe a way to handle this is to check `self.holder_commitment_point.is_available()` with our checks above before we start updating state, then when we get here it's easier to know that this is unreachable?

Reply:
or maybe just move this to be the first thing we update?

Reply:
I don't feel strongly since Matt clarified that this should be unreachable. Could also update the comment to be more explicit about that, or what you suggested, up to you!

Reply:
just updated the last sentence of the comment to clarify this should be unreachable